modules/app-config/index.tsx

```diff
@@ -0,0 +1,35 @@
+import {getFeatureFlag} from '../../source/lib/storage'
+import {AppConfigKey, AppConfigGroupKey, FeatureFlagType} from './types'
+import {useQuery} from '@tanstack/react-query'
+import {isDevMode} from '@frogpond/constants'
+
+export type {AppConfigKey, FeatureFlagType} from './types'
+
+// datastore for the __DEV__ feature flags
+export const AppConfig = async (): Promise<FeatureFlagType[]> => {
+	return [
+		{
+			configKey: AppConfigKey.TestDataKey,
+			group: AppConfigGroupKey.test,
+			title: 'Test config key',
+			active: await getFeatureFlag(AppConfigKey.TestDataKey),
+		},
+	]
+}
+
+// helper method to query exported __DEV__ feature flags
+const useFeature = (featureKey: AppConfigKey): boolean => {
+	let {data: featureValue = false} = useQuery({
+		queryKey: ['app', 'app:feature-flag', featureKey],
+		queryFn: () => getFeatureFlag(featureKey),
+		onSuccess: (newValue) => {
+			return isDevMode() ? newValue : false
+		},
+	})
+
+	return isDevMode() ? featureValue : false
+}
+
+// exported feature flags
+export const useTestFeature = (): boolean =>
+	useFeature(AppConfigKey.TestDataKey)
```

Comment:
Is there any way we can add some tests for all this? Seems like some relatively contained logic.

Reply:
I almost didn't open the PR until I wrote some tests knowing I'd see this comment. I definitely will test this.

Reply:
@hawkrives @rye are we wanting to add `@testing-library/react-hooks` and `react-test-renderer` to the repo? To test the hooks behind `useFeature` [would be to test  tanstack query.](https://tanstack.com/query/v4/docs/react/guides/testing)

Reply:
I am happy with us adding those, if that's what you're asking.

Are we above those versions for `@testing-library/react-hooks@8.0.1`? or below?

Reply:
I think I figured it out. 

> As part of the changes for React 18, it has been decided that the renderHook API provided by this library will instead be included as official additions to both `react-testing-library` and `react-native-testing-library` with the intention being to provide a more cohesive and consistent implementation for our users.

I had to add `@testing-library/react-native` instead, which has the functionality we need from `@testing-library/react`. And for future note,  `@testing-library/react-hooks` is limited to react 17 but supports 18 now that it is absorbed into its parent package.

Reply:
Getting closer! Dealing with issues when running the tests. Inconsistent results, sometimes passing, sometimes tanstack errors.

```
ReferenceError: You are trying to `import` a file after the Jest environment has been torn down.

at batchNotifyFn (node_modules/react-native/Libraries/ReactNative/RendererImplementation.js:44:10)
at node_modules/@tanstack/query-core/src/notifyManager.ts:65:9
node_modules/react-native/Libraries/ReactNative/RendererImplementation.js:44

TypeError: require(...).unstable_batchedUpdates is not a function
    at batchNotifyFn (node_modules/react-native/Libraries/ReactNative/RendererImplementation.js:44:51)
    atnode_modules/@tanstack/uery-core/src/notifyManager.ts:65:9
```